cfme/common/provider_views.py

```diff
@@ -524,6 +545,18 @@ def is_displayed(self):
                 self.title.text == 'Edit Infrastructure Provider')
 
 
+class PhysicalProviderEditView(ProviderEditView):
+    """
+     represents Provider Edit View
+    """
+    @property
+    def is_displayed(self):
+        return (super(PhysicalProviderEditView, self).is_displayed and
+                self.navigation.currently_selected ==
+                ['Compute''Physical Infrastructure', 'Providers'] and
+                self.title.text == 'Edit Infrastructure Provider')
```

Comment:
I don't think this is sufficient for `is_displayed` check, because this only validates that we're on *an* Edit page, not the edit page for a specific provider object. Need to include a check against `self.context['object'].name` here to verify we're on the edit page for a specific provider - or return False if this isn't possible (There are several pages like this in CFME where we have to return False and always navigate)

Reply:
I understand.